test/test_known_hosts_cached.rb.rb

```diff
@@ -0,0 +1,202 @@
+require 'common'
+require 'net/ssh/known_hosts/cached'
+
+class TestKnownHostsCached < NetSSHTest
+  def build_opts(user_file, global_file = 'does_not_exist')
+    {
+      user_known_hosts_file: user_file,
+      global_known_hosts_file: global_file
+    }
+  end
+
+  def perform_test(source)
+    # For these tests we only want to read a single file, set that as 
+    # the user_known_hosts_file and pass a non-existant file to 
+    # global_known_hosts_file
+    kh = Net::SSH::KnownHosts::Cached.new(build_opts(source))
+    keys = kh.search_for("github.com")
+    assert_equal(1, keys.count)
+    assert_equal("ssh-rsa", keys.first.ssh_type)
+  end
+
+  def test_key_for_when_all_hosts_are_recognized
+    perform_test(path("known_hosts/github"))
+  end
+
+  def test_key_for_when_an_host_hash_is_recognized
+    perform_test(path("known_hosts/github_hash"))
+  end
+
+  def test_parsing_known_hosts_empty_lines_and_comments
+    perform_test(path("known_hosts/known_hosts_ignore"))
+  end
+
+  def test_missing_then_add
+    Tempfile.open('github') do |f|
+      f.write(File.read(path("known_hosts/github")))
+      kh = Net::SSH::KnownHosts::Cached.new(build_opts(f.path))
+      keys = kh.search_for("github2.com")
+      assert_equal(0, keys.count)
+      assert_equal([], keys.to_a)
+
+      kh.add('github2.com',rsa_key)
+      keys2 = kh.search_for("github2.com")
+      assert_equal([rsa_key.to_blob], keys2.to_a.map(&:to_blob))
+    end
+  end
+
+  def 
```

Comment:
remove this

Reply:
Wow, yeah that was an interesting mistake. That was effectively causing that test to be skipped. I'm surprised that did not cause any warnings or errors. :man_shrugging: fixed.